core/src/test/java/io/atomix/core/lock/impl/DistributedLockTest.java

```diff
@@ -63,9 +65,46 @@ public void testTryLockFail() throws Throwable {
 
     lock1.lock().join();
 
+    assertFalse(lock2.tryLock().join().isPresent());
+  }
+
+  /**
+   * Tests attempting to acquire a lock.
+   */
+  @Test
+  public void testTryLockSucceed() throws Throwable {
+    AsyncDistributedLock lock = atomix().lockBuilder("test-try-lock-succeed").build().async();
+    assertTrue(lock.tryLock().join().isPresent());
+  }
+
+  /**
+   * Tests attempting to acquire a lock with a timeout.
+   */
+  @Test
+  public void testTryLockFailWithTimeout() throws Throwable {
+    AsyncDistributedLock lock1 = atomix().lockBuilder("test-try-lock-fail-with-timeout").build().async();
+    AsyncDistributedLock lock2 = atomix().lockBuilder("test-try-lock-fail-with-timeout").build().async();
+
+    lock1.lock().join();
+
     assertFalse(lock2.tryLock(Duration.ofSeconds(1)).join().isPresent());
   }
 
+  /**
+   * Tests attempting to acquire a lock with a timeout.
+   */
+  @Test
+  public void testTryLockSucceedWithTimeout() throws Throwable {
+    AsyncDistributedLock lock1 = atomix().lockBuilder("test-try-lock-succeed-with-timeout").build().async();
+    AsyncDistributedLock lock2 = atomix().lockBuilder("test-try-lock-succeed-with-timeout").build().async();
+
+    lock1.lock().join();
+
+    CompletableFuture<Optional<Version>> future = lock2.tryLock(Duration.ofSeconds(1));
+    lock1.unlock().join();
+    assertTrue(future.join().isPresent());
+  }
+
   /**
    * Tests unlocking a lock with a blocking call in the event thread.
```

Comment:
Should we also assert is present for the joins in testBlockingUnlock?

Reply:
yep

Reply:
Reviewing from mobile ftl :d